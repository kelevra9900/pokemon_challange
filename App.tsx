import React from 'react';
import 'react-native-gesture-handler';
import {NavigationContainer} from '@react-navigation/native';
import {Navigator} from './src/navigator/Navigator';
import {AuthProvider} from './src/context/AuthContext';
import {Provider as PaperProvider} from 'react-native-paper';

const AppState = ({children}: any) => {
  return <AuthProvider>{children}</AuthProvider>;
};

const App = () => {
  return (
    <NavigationContainer>
      <AppState>
        <PaperProvider>
          <Navigator />
        </PaperProvider>
      </AppState>
    </NavigationContainer>
  );
};

export default App;
