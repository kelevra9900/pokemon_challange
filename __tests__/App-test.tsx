/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// const mockedNavigate = jest.fn();

// jest.mock('@react-navigation/native', () => {
//   const actualNav = jest.requireActual('@react-navigation/native');
//   return {
//     ...actualNav,
//     useNavigation: () => ({
//       navigate: mockedNavigate,
//     }),
//   };
// });

describe('When Login render', () => {
  it('renders correctly', () => {
    renderer.create(<App />).toJSON();
    expect(renderer.create(<App />)).toMatchSnapshot();
  });
});
