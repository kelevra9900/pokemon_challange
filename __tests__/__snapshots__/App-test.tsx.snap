// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Login render renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={2}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        Object {
                          "backgroundColor": "#f8f8f8",
                        },
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#111111",
                            "height": 1200,
                            "position": "absolute",
                            "top": -250,
                            "width": 1000,
                          }
                        }
                      />
                      <View
                        onLayout={[Function]}
                        style={
                          Array [
                            Object {
                              "flex": 1,
                            },
                            Object {
                              "paddingBottom": 0,
                            },
                          ]
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Object {
                              "flex": 1,
                              "height": 600,
                              "justifyContent": "center",
                              "marginBottom": 50,
                              "marginHorizontal": 16,
                              "paddingHorizontal": 20,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "flex": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#F3F1F3",
                                  "borderRadius": 6,
                                  "elevation": 5,
                                  "flexDirection": "row",
                                  "height": 40,
                                  "marginTop": 20,
                                  "paddingHorizontal": 20,
                                  "shadowColor": "#000",
                                  "shadowOffset": Object {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                }
                              }
                            >
                              <TextInput
                                autoCapitalize="none"
                                autoCorrect={false}
                                keyboardType="email-address"
                                placeholder="Email"
                                placeholderTextColor="#828282"
                                style={
                                  Array [
                                    Object {
                                      "fontSize": 20,
                                    },
                                    Object {
                                      "fontSize": 16,
                                      "marginLeft": 6,
                                      "paddingBottom": 4,
                                    },
                                  ]
                                }
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "#F3F1F3",
                                  "borderRadius": 6,
                                  "elevation": 5,
                                  "flexDirection": "row",
                                  "height": 40,
                                  "marginTop": 20,
                                  "paddingHorizontal": 20,
                                  "shadowColor": "#000",
                                  "shadowOffset": Object {
                                    "height": 2,
                                    "width": 0,
                                  },
                                  "shadowOpacity": 0.25,
                                  "shadowRadius": 3.84,
                                }
                              }
                            >
                              <TextInput
                                autoCapitalize="none"
                                autoCorrect={false}
                                placeholder="Password"
                                placeholderTextColor="#828282"
                                secureTextEntry={true}
                                style={
                                  Array [
                                    Object {
                                      "fontSize": 20,
                                    },
                                    Object {
                                      "fontSize": 16,
                                      "marginLeft": 6,
                                      "paddingBottom": 4,
                                    },
                                  ]
                                }
                              />
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "marginTop": 30,
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={false}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "backgroundColor": "#2F80ED",
                                    "borderRadius": 8,
                                    "height": 48,
                                    "justifyContent": "center",
                                    "opacity": 1,
                                    "paddingHorizontal": 20,
                                    "paddingVertical": 5,
                                    "width": "100%",
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "white",
                                      "fontSize": 16,
                                    }
                                  }
                                >
                                  Log In
                                </Text>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "marginTop": 30,
                                }
                              }
                            >
                              <View
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Object {
                                      "color": "#2F80ED",
                                      "fontSize": 16,
                                      "textDecorationLine": "underline",
                                    }
                                  }
                                >
                                  Create an account
                                </Text>
                              </View>
                            </View>
                          </View>
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
